src/main/kotlin/uk/gov/gdx/datashare/services/EventDataService.kt

```diff
@@ -43,11 +46,16 @@ class EventDataService(
     id: UUID,
   ): EventNotification {
     val clientId = authenticationFacade.getUsername()
-    val event = eventDataRepository.findByClientIdAndId(clientId, id)
+    val event = acquirerEventRepository.findByClientIdAndId(clientId, id)
       ?: throw EventNotFoundException("Event $id not found for polling client $clientId")
     val acquirerSubscription = acquirerSubscriptionRepository.findByEventId(id)
       ?: throw AcquirerSubscriptionNotFoundException("Acquirer subscription not found for event $id")
-    val enrichmentFieldNames = enrichmentFieldNamesForAcquirerSubscription(acquirerSubscription)
+
+    return buildEnrichedEventNotification(event, acquirerSubscription)
+  }
+
+  fun buildEnrichedEventNotification(event: AcquirerEvent, acquirerSubscription: AcquirerSubscription): EventNotification {
```

Comment:
can be private

Reply:
Will do for now, but the next PR will make it public again